thirtyinch-kotlin-coroutines/src/main/java/net/grandcentrix/thirtyinch/kotlin/coroutines/TiCoroutineScope.kt

```diff
@@ -0,0 +1,79 @@
+package net.grandcentrix.thirtyinch.kotlin.coroutines
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.launch
+import net.grandcentrix.thirtyinch.TiPresenter
+import net.grandcentrix.thirtyinch.TiPresenter.State.DESTROYED
+import net.grandcentrix.thirtyinch.TiPresenter.State.VIEW_ATTACHED
+import net.grandcentrix.thirtyinch.TiPresenter.State.VIEW_DETACHED
+import kotlin.coroutines.CoroutineContext
+
+/**
+ * A [CoroutineScope] that is bound to the lifecycle of the given [TiPresenter].
+ *
+ * @param cancelWhenViewDetaches If all started coroutines in this scope should be cancelled when the view detaches.
+ * By default they're cancelled when `presenter` is destroyed. You should recreate this object when the view attaches,
+ * if you set this to `true`.
+ */
+class TiCoroutineScope(
+        private val presenter: TiPresenter<*>,
+        private val context: CoroutineContext
+) : CoroutineScope {
+
+    /**
+     * Parent [Job] for all coroutines that are started in the given presenter instance. When this `job` gets
+     * cancelled all of its children jobs will get cancelled too.
+     */
+    private val job = Job()
+    override val coroutineContext: CoroutineContext = context + job
+    /**
+     * Parent [Job] for all coroutines that are started while a `view` attached. When this `job` gets
+     * cancelled (when the view detaches) all of its children jobs will get cancelled too.
+     */
+    private var onViewDetachJob: Job? = null
```

Comment:
This Job is named (and documented) better as the "job" above.
I think it would be good to rename the `job` above to `presenterDestroyJob` and add the doc that this job will be destroyed when the presenter got destroyed...